Source/SwiftLintFramework/Rules/Style/NumberSeparatorRule.swift

```diff
@@ -108,6 +115,14 @@ public struct NumberSeparatorRule: OptInRule, CorrectableRule, ConfigurationProv
         return prefixes.filter { lowercased.hasPrefix($0) }.isEmpty
     }
 
+    private func isInValidRanges(number: String) -> Bool {
+        if let double = Double(number), configuration.validRanges.contains(where: { $0.contains(double) }) {
```

Comment:
I don't think `Double(number)` works if there's a `_`.

Reply:
Fixed in a8d06ef162f79bb444e473b9720f3f9708bd91d7.